samza-api/src/main/java/org/apache/samza/operators/descriptors/base/stream/InputDescriptor.java

```diff
@@ -73,16 +73,15 @@ public InputDescriptor(String streamId, Serde serde, SystemDescriptor systemDesc
   }
 
   /**
-   * If set to true, when a Samza container starts up, it ignores any checkpointed offset for this particular
+   * If set, when a Samza container starts up, it ignores any checkpointed offset for this particular
    * input stream. Its behavior is thus determined by the {@link #withOffsetDefault} setting.
    * Note that the reset takes effect every time a container is started, which may be every time you restart your job,
    * or more frequently if a container fails and is restarted by the framework.
    *
-   * @param resetOffset whether the container should ignore any checkpointed offset when starting
    * @return this input descriptor
    */
-  public SubClass withResetOffset(boolean resetOffset) {
```

Comment:
I do like the look of this proposed API better, but this is kind of now coupling with the fact that the default is "false", since this can't set it to false anymore. Is it a reasonable assumption that these all always will default to false? Or will someone need to remember to update this if the default changes?

Reply:
Changing a default value would be a backwards incompatible change, so making a corresponding API change should be OK. In general going forward we will need to keep the configuration and descriptors in sync (including javadocs etc.).